### PR TITLE
Fix #8692: fix "tsserver as a library" (lssl) build target

### LINKED ISSUE #8692: Default tsserver behaviour cannot be changed
Hi. I am working on an integration with tsserver and there is a little problem: I need to override a default behaviour for IOSession but I cannot do it because:
1) IOSession and ioSession are private
2) a listening session is always started ([see here](https://github.com/Microsoft/TypeScript/blob/a09b001c6180b39155832b2f18be708ea92da263/src/server/server.ts#L278-L278))

I have to use my own tsserver implementation as a workaround.

Possible solution: check an environment flag before running ioSession.listen().

### PR DESCRIPTION
**Pull request by mhegazy**
Fixes #8692

### CODE REVIEW

**Comment by mhegazy** [comment]
@zhengbli can you take a look

File: tests/cases/unittests/cachingInServerLSHost.ts [hunk 1]
```diff
@@ -54,12 +54,18 @@ module ts {
...
+                };
             },
             watchDirectory: (path, callback, recursive?) => {
                 return {
                     close: () => { }
-                }
+                };
+            },
+            setTimeout: (callback, ms, ...args) => {
+                return setTimeout(callback, ms, args);
```
Comment (zhengbli) [inline tests/cases/unittests/cachingInServerLSHost.ts]: Why not just `{ ..., setTimeout, clearTimeout, ... }`

**Comment by zhengbli** [comment]
As discussed the line https://github.com/Microsoft/TypeScript/blob/master/src/server/editorServices.ts#L1383 might need to be changed as well. Otherwise LGTM. :+1:

File: tests/cases/unittests/cachingInServerLSHost.ts (hunk 1 above)
Comment (mhegazy) [inline tests/cases/unittests/cachingInServerLSHost.ts]: ahh. nice.

**Pull request merged**